src/decode.rs

```diff
@@ -4364,22 +4362,15 @@ pub(crate) unsafe fn rav1d_decode_frame_init(
             }
 
             let pal_idx_sz = num_sb128 * size_mul[1] as c_int;
-            if pal_idx_sz != f.frame_thread.pal_idx_sz {
-                rav1d_freep_aligned(&mut f.frame_thread.pal_idx as *mut *mut u8 as *mut c_void);
-                f.frame_thread.pal_idx = rav1d_alloc_aligned(
-                    ::core::mem::size_of::<u8>() * pal_idx_sz as usize * 128 * 128 / 4,
-                    64,
-                ) as *mut u8;
-                if f.frame_thread.pal_idx.is_null() {
-                    f.frame_thread.pal_idx_sz = 0;
-                    return Err(ENOMEM);
-                }
-                f.frame_thread.pal_idx_sz = pal_idx_sz;
+            if pal_idx_sz as usize != f.frame_thread.pal_idx_sz() {
+                // TODO: Fallible allocation
+                f.frame_thread
+                    .pal_idx
+                    .resize_with(pal_idx_sz as usize * 128 * 128 / 4, Default::default);
             }
```

Comment:
```suggestion
            // TODO: Fallible allocation
            f.frame_thread
                .pal_idx
                .resize_with(pal_idx_sz as usize * 128 * 128 / 4, Default::default);
```

`.resize_with` will be a no-op when the sizes are equal.  I think what the C was trying to do is similar to what we're doing here with `.resize_with`, so don't have to duplicate that part.